lib/core/text/icu/caseModification.nit

```diff
@@ -0,0 +1,181 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Adding locale aware case modification to the String and CString classes
+
+module caseModification is pkgconfig("icu-io", "icu-i18n", "icu-uc")
+import locale
+
+`{
+	#include <unicode/utypes.h>
+	#include <unicode/ustring.h>
+
+	#include <string.h>
+	#include <locale.h>
+
+`}
+
+
+
+# There are 3 different types of case modification but they all behave the same way
+# Therefore the main operations are in a parent class
+private abstract class CaseModifier
+
+	fun getModifiedStrLength (uchar_str, error : Pointer, strLength : Int, localeName : CString) : Int  is abstract
+	fun doModification  (uchar_modifiedStr, uchar_str, error : Pointer, requiredLength, strLength : Int, localeName : CString) is abstract
+
+	fun modify (locale : Locale, str : CString) : CString import String.to_cstring, CString.cstring_length, Locale.getName, getModifiedStrLength, doModification `{
```

Comment:
callbacks are fun and nice but the current approach are to avoid them:

* the interpreter cannot execute them (because reasons)
* they tend to make a blurry line between the nit world (beautiful and pure, and the real C world full of pettiness and gore)

one way is to split function into two parts: al nitty one, and a native one

~~~nit
redef class String
   fun do_magic: String do
      return to_cstring.do_magic.to_s
   end
end

redef class CString
   private fun do_magic: CString `{
      self[0] = 'A'; // FIXME this is ugly and unsafe and useless
      return self;
   `}
end

print "bbbb".do_magic
~~~

Maybe the doc should be updated with current good practices @xymus